File: WORKSPACE/code/00_setup/00_config.py
# 00_config.py -- baseline project configuration
from __future__ import annotations
import os
from pathlib import Path


def _find_project_root(start: Path = Path.cwd(), max_up: int = 10) -> Path:
    cur = start.resolve()
    for _ in range(max_up + 1):
        if (cur / "WORKSPACE" / "code" / "run_all.py").exists() or \
           (cur / "WORKSPACE" / "code" / "run_all.R").exists() or \
           (cur / "WORKSPACE" / "code" / "run_all.do").exists():
            return cur
        parent = cur.parent
        if parent == cur:
            break
        cur = parent
    raise RuntimeError(
        "Could not locate project root. "
        "Set EIG_PROJECT_ROOT or run from the repo root."
    )


# Project root: env var takes priority, then walk upward from this file
_env_root = os.environ.get("EIG_PROJECT_ROOT", "")
PATH_PROJECT = Path(_env_root).resolve() if _env_root else _find_project_root(
    Path(__file__).resolve().parent
)

# Paths
PATH_CODE                = PATH_PROJECT / "WORKSPACE" / "code"
PATH_DATA                = PATH_PROJECT / "WORKSPACE" / "data"
PATH_DATA_RAW            = PATH_DATA / "raw"
PATH_DATA_PROCESSED      = PATH_DATA / "processed"
PATH_OUTPUT              = PATH_PROJECT / "WORKSPACE" / "output"
PATH_OUTPUT_FIG_MAIN     = PATH_OUTPUT / "figures" / "main"
PATH_OUTPUT_FIG_APPENDIX = PATH_OUTPUT / "figures" / "appendix"
PATH_OUTPUT_TBL_MAIN     = PATH_OUTPUT / "tables" / "main"
PATH_OUTPUT_TBL_APPENDIX = PATH_OUTPUT / "tables" / "appendix"
PATH_OUTPUT_INTERMEDIATE = PATH_OUTPUT / "data" / "intermediate_results"

# Project settings
cfg = {
    "project_name":       "eig-wagesubsidy-policy-sim",
    "audience":           "Public — interactive web simulation + blog post",
    "project_scope_tier": 1,   # 1=Descriptive/Blog  2=Analytical Brief  3=Full Research Paper
    "currency_base_year": 2024,
    "fig_width":          6.5,
    "fig_height":         3.5,
    "fig_dpi":            300,
    "seed":               1234,
    # Wage subsidy policy parameters (EIG 80-80 Rule)
    "ws_median_hourly_wage":  20.00,  # 2024 CPS median hourly wage for hourly workers
    "ws_target_pct":          0.80,   # target wage = 80% of median
    "ws_subsidy_pct":         0.80,   # subsidy covers 80% of gap
    "ws_base_wage":           7.25,   # federal minimum wage (base wage floor)
    # Derived (convenience)
    "ws_target_wage":         16.00,  # 0.80 * 20.00
    "ws_max_subsidy":         7.00,   # 0.80 * (16.00 - 7.25) ≈ 7.00
    "ws_hours_per_year":      2000,   # 40 hrs/wk * 50 wks
}

# Directory creation — base dirs (all tiers)
for _d in [PATH_DATA_RAW, PATH_DATA_PROCESSED, PATH_OUTPUT_FIG_MAIN, PATH_OUTPUT_TBL_MAIN]:
    _d.mkdir(parents=True, exist_ok=True)

# Tier 2+: intermediate results
if cfg["project_scope_tier"] >= 2:
    PATH_OUTPUT_INTERMEDIATE.mkdir(parents=True, exist_ok=True)

# Tier 3: appendix directories
if cfg["project_scope_tier"] >= 3:
    PATH_OUTPUT_FIG_APPENDIX.mkdir(parents=True, exist_ok=True)
    PATH_OUTPUT_TBL_APPENDIX.mkdir(parents=True, exist_ok=True)

print(f"Loaded config. Project root: {PATH_PROJECT}")

File: WORKSPACE/code/00_setup/00_config.R
# 00_config.R -- baseline project configuration
options(scipen = 999)
options(dplyr.summarise.inform = FALSE)

find_project_root <- function(start = getwd(), max_up = 10L) {
  cur <- normalizePath(start, winslash = "/", mustWork = FALSE)
  for (i in seq_len(max_up + 1L)) {
    if (file.exists(file.path(cur, "WORKSPACE", "code", "run_all.R"))) return(cur)
    parent <- normalizePath(file.path(cur, ".."), winslash = "/", mustWork = FALSE)
    if (identical(parent, cur)) break
    cur <- parent
  }
  NA_character_
}

path_project <- Sys.getenv("EIG_PROJECT_ROOT", unset = "")
if (!nzchar(path_project)) path_project <- find_project_root()
if (is.na(path_project) || !dir.exists(path_project)) {
  stop("Could not locate project root. Set EIG_PROJECT_ROOT or run from repo.", call. = FALSE)
}

path_project <- normalizePath(path_project, winslash = "/", mustWork = TRUE)
setwd(path_project)

path_code <- file.path(path_project, "WORKSPACE", "code")
path_data <- file.path(path_project, "WORKSPACE", "data")
path_data_raw <- file.path(path_data, "raw")
path_data_processed <- file.path(path_data, "processed")
path_output <- file.path(path_project, "WORKSPACE", "output")
path_output_fig_main <- file.path(path_output, "figures", "main")
path_output_fig_appendix <- file.path(path_output, "figures", "appendix")
path_output_tbl_main <- file.path(path_output, "tables", "main")
path_output_tbl_appendix <- file.path(path_output, "tables", "appendix")
path_output_intermediate <- file.path(path_output, "data", "intermediate_results")

cfg <- list(
  project_name = "[PROJECT_NAME]",
  audience = "[AUDIENCE]",
  project_scope_tier = 3L,  # 1=Descriptive/Blog, 2=Analytical Brief, 3=Full Research Paper
  currency_base_year = 2025L,
  fig_width = 10,
  fig_height = 6,
  fig_dpi = 300,
  seed = 1234
)

# Base directories required at all tiers
base_dirs <- c(path_data_raw, path_data_processed, path_output_fig_main, path_output_tbl_main)
for (d in base_dirs) dir.create(d, recursive = TRUE, showWarnings = FALSE)

# Intermediate results directory (always created; stage contracts reference it at all tiers)
dir.create(path_output_intermediate, recursive = TRUE, showWarnings = FALSE)

# Appendix output directories (tier 3 only)
if (cfg$project_scope_tier >= 3L) {
  dir.create(path_output_fig_appendix, recursive = TRUE, showWarnings = FALSE)
  dir.create(path_output_tbl_appendix, recursive = TRUE, showWarnings = FALSE)
}

files <- list(
  run_manifest = file.path(path_output_intermediate, "run_manifest.json"),
  data_report = file.path(path_project, "INFRA", "quality_reports", "data_report.html")
)

set.seed(cfg$seed)
message("Loaded config. Project root: ", path_project)
